cmd/trickster/config.go

```diff
@@ -219,6 +219,12 @@ func applyCachingConfig(c, oc *config.Config, logger *log.Logger,
 				caches[k] = w
 				continue
 			}
+
+			// if we got to this point, the cache won't be used, so lets close it
+			go func() {
+				time.Sleep(time.Second * time.Duration(c.ReloadConfig.DrainTimeoutSecs))
+				w.Close()
```

Comment:
Is this threadsafe?

Reply:
In the worst case, it will throw an inconsequential panic on the no-longer-used cache. But point taken and we'll add locking around the entire reload handler in a subsequent PR.